src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1249,39 +1249,76 @@ private void ImportThrow()
         {
         }
 
+        private LLVMValueRef GetInstanceFieldAddress(StackEntry objectEntry, FieldDesc field)
+        {
+            var objectType = objectEntry.Type ?? field.OwningType;
+            LLVMValueRef typedObjectValue;
+            if (objectType.IsValueType && !objectType.IsPointer && objectEntry.Kind != StackValueKind.NativeInt)
+            {
+                typedObjectValue = LLVM.BuildAlloca(_builder, GetLLVMTypeForTypeDesc(objectType), "objptr");
```

Comment:
Why are you copying the object here?

Reply:
if the LLVMValueRef isnt a pointer, I was generally unsuccessful in trying to either cast it to one or take an address. When I looked at how this opcode is implemented in https://github.com/SharpNative/CSharpLLVM/blob/13f69af978443924cf3f029c40b569a89e87c28f/CSharpLLVM/Generator/Instructions/Objects/EmitLdfld.cs it seemed like there wasn't another way to solve it.